app/services/service_provider_updater.rb

```diff
@@ -11,21 +11,26 @@ class ServiceProviderUpdater
     cert
   ]
 
-  def run
-    dashboard_service_providers.each do |service_provider|
+  def run(service_provider = nil)
+    if service_provider.present?
       update_local_caches(ActiveSupport::HashWithIndifferentAccess.new(service_provider))
+    else
+      dashboard_service_providers.each do |dashboard_service_provider|
+        update_local_caches(ActiveSupport::HashWithIndifferentAccess.new(dashboard_service_provider))
+      end
     end
   end
 
   private
 
   def update_local_caches(service_provider)
-    issuer = service_provider['issuer']
-    update_cache(issuer, service_provider)
+    update_cache(service_provider)
   end
 
-  def update_cache(issuer, service_provider)
-    if service_provider['active'] == true
+  def update_cache(service_provider)
+    issuer = service_provider['issuer']
+    # when a service provider is passed in via params, true becomes 'true'
+    if [true, 'true'].include?(service_provider['active'])
```

Comment:
are we doing a form post?

if we switched to JSON encoding of the body when we post, we'd likely preserve boolean `true`

Reply:
ah thanks, fixed in [`8ac09c9` (#9032)](https://github.com/18F/identity-idp/pull/9032/commits/8ac09c9bcd86728a1299b81e81278ad474fcb04b)

Reply:
Rails will automatically parse JSON when entire POST bodies are JSON (if the `Content-Type` header is set correctly) so I don't think we need to do that

Reply:
hm ok i hadn't updated the headers of the original request, i'll do that and see if it works!

Reply:
fixed here [`2be634b` (#9032)](https://github.com/18F/identity-idp/pull/9032/commits/2be634bc391ad16ceac21f23897f491424663c81) and on the dashboard here: [18F/identity-dashboard@`c4aa305` (#663)](https://github.com/18F/identity-dashboard/pull/663/commits/c4aa30579ddde27e0fc6a6bf8aa49d0b9956f13e)